src/Concerns/HasCallback.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace LaraCrafts\GeoRoutes\Concerns;
+
+use Illuminate\Support\Str;
+
+trait HasCallback
+{
+    /**
+     * The callback to execute if the visitor
+     * is not allowed.
+     *
+     * @var array
+     */
+    protected $callback;
+
+    /**
+     * The callbacks' proxies.
+     *
+     * @var array
+     */
+    protected static $proxies;
+
+    /**
+     * Load the available proxies.
+     */
+    protected static function loadProxies()
+    {
+        if (static::$proxies !== null) {
+            return;
+        }
+
+        static::$proxies = [];
+        $callbacks = config('geo-routes.routes.callbacks');
+
+        foreach ($callbacks as $key => $callback) {
+            static::$proxies['or' . Str::studly($key)] = $callback;
+        }
+    }
+
+    /**
+     * Return a HTTP 404 error if access is denied.
+     *
+     * @return $this
+     */
+    public function orNotFound()
+    {
+        return $this->setCallback('LaraCrafts\GeoRoutes\Callbacks::notFound', func_get_args());
+    }
+
+    /**
+     * Redirect to given route if access is denied.
+     *
+     * @param string $routeName
+     *
+     * @return $this
+     */
+    public function orRedirectTo(string $routeName)
+    {
+        return $this->setCallback('LaraCrafts\GeoRoutes\Callbacks::redirectTo', func_get_args());
+    }
+
+    /**
+     * Return a HTTP 401 error if access is denied (this is the default behavior).
+     *
+     * @return $this
+     */
+    public function orUnauthorized()
+    {
+        $this->callback = null;
+
+        return $this;
+    }
+
+    /**
+     * Set the callback.
+     *
+     * @param callable $callback
+     * @param array $arguments
+     *
+     * @return $this
+     */
+    protected function setCallback(callable $callback, array $arguments)
+    {
+        if (is_string($callback) && Str::contains($callback, '@')) {
+            $callback = Str::parseCallback($callback, '__invoke');
+            $callback[0] = resolve($callback[0]);
```

Comment:
If I’m not mistaken this means the callback is resolved as soon as the route is defined, would it not be better to “lazy” load it when its actually needed (a geo route gets invoked by the router). Because right now having a lot of routes with a particularly heavy callback class might negatively impact framework performance

Reply:
This is actually not the topic of this *PR* as the presented piece of code is currently unchanged, i think a new pull request should cover the suggested changes.